File: Library-Borrowing-System-main/src/services/userService.js
import api from '../utils/api'

export function login(data) {
  return api.post('/users/login', data)
}

export function register(data) {
  return api.post('/users/register', data)
}

export function getProfile() {
  return api.get('/users/profile')
}

export function getAllUsers(page, size) {
  return api.get('/users/getAllUsers', { params: { page, size } })
}

export function getUserById(id) {
  return api.get(`/users/getUserById/${id}`)
}

export function getUserByUsername(username) {
  return api.get(`/users/getUserByUsername/${username}`)
}

export function updateUser(id, data) {
  return api.put(`/users/updateUser/${id}`, data)
}

export function deleteUser(id) {
  return api.delete(`/users/deleteUser/${id}`)
}

export function getCaptcha() {
  return api.get('/users/captcha')
}

export function verifyCaptcha(data) {
  return api.post('/users/captcha/verify', data)
}

export function getDeletedUsers(page, size) {
  return api.get('/users/recycleBin', { params: { page, size } })
}

export function restoreUser(id) {
  return api.put(`/users/restore/${id}`)
}

export function cleanExpiredUsers(retentionDays = 30) {
  return api.delete('/users/cleanExpired', { params: { retentionDays } })
}
// 用户自己改密码
export function changePassword(data) {
  return api.put('/users/password', data)
}

// 管理员重置密码
export function resetPassword(id, data) {
  return api.put(`/users/${id}/password`, data)
}

File: Library-Borrowing-System-main/server/src/main/java/net/togogo/service/UserService.java
package net.togogo.service;

import net.togogo.dto.ChangePasswordRequest;
import net.togogo.dto.LoginRequest;
import net.togogo.dto.LoginResponse;
import net.togogo.dto.PageResponse;
import net.togogo.dto.RegisterRequest;
import net.togogo.dto.ResetPasswordRequest;
import net.togogo.dto.UpdateUserRequest;
import net.togogo.dto.UserDTO;
import org.springframework.data.domain.Pageable;

import java.util.Map;

public interface UserService {
    UserDTO register(RegisterRequest request);
    LoginResponse login(LoginRequest request, String ip);
    PageResponse<UserDTO> getAllUsers(Pageable pageable);
    UserDTO getUserById(Long id);
    UserDTO getUserByUsername(String username);
    UserDTO getUserByPhone(String phone);
    void deleteUser(Long id);
    UserDTO updateUser(Long id, UpdateUserRequest request);

    Map<String, String> generateCaptcha();
    void verifyCaptcha(String captchaKey, String inputCaptcha);
    UserDTO restoreUser(Long id);
    PageResponse<UserDTO> getDeletedUsers(Pageable pageable);
    void cleanExpiredUsers(int retentionDays);
    /**
    * 当前登录用户修改自己的密码（需验证原密码）
     */
    void changePassword(ChangePasswordRequest request);
    /**
     * 管理员重置指定用户的密码（无需原密码）
     */
    void resetPassword(Long userId, ResetPasswordRequest request);
}
